app/src/main/java/me/hufman/androidautoidrive/music/controllers/SpotifyAppController.kt

```diff
@@ -244,19 +248,80 @@ class SpotifyAppController(context: Context, val remote: SpotifyAppRemote, val w
 	 * not authorized then the the [QueueMetadata] is created from the app remote API.
 	 */
 	fun createLikedSongsQueueMetadata() {
+		if (createQueueMetadataJob?.isActive == true) {
+			createQueueMetadataJob?.cancel()
+		}
+
 		createQueueMetadataJob = GlobalScope.launch(defaultDispatcher) {
 			queueItems = webApi.getLikedSongs(this@SpotifyAppController) ?: emptyList()
+
 			if (queueItems.isNotEmpty()) {
-				queueMetadata = QueueMetadata("Liked Songs", null, queueItems)
-				loadQueueCoverart()
+				queueMetadata = QueueMetadata(L.MUSIC_LIKED_SONGS_PLAYLIST_NAME, null, queueItems)
+
+				val hashCode = queueItems.hashCode().toString()
+				val likedSongsStateJson = appSettings[AppSettings.KEYS.SPOTIFY_LIKED_SONGS_PLAYLIST_STATE]
+				val likedSongsState: LikedSongsState
+				if (likedSongsStateJson.isBlank()) {
+					Log.d(TAG, "No previous liked songs state found.")
+					val uri = webApi.createPlaylist(SpotifyWebApi.LIKED_SONGS_PLAYLIST_NAME)
+					if (uri == null) {
+						Log.e(TAG, "Error creating liked songs playlist, falling back to app remote API")
+						queueItems = emptyList()
```

Comment:
There should be a comment here indicating that we return an empty list to prevent trying the broken skipToIndex behavior in https://github.com/spotify/android-sdk/issues/267

Reply:
With this the reasoning behind the emptyList is actually not to prevent the creation of the Liked Songs playlist with any data as we still want this broken behavior since at the very least the user can still access their Liked Songs playlist even if the skipToIndex logic doesn't work as it should. Instead this is so that the createQueueMetadata will properly create the Liked Songs playlist as this conditional will fail otherwise https://github.com/Ben2356/AndroidAutoIdrive/blob/spotifyLikedSongsSkip/app/src/main/java/me/hufman/androidautoidrive/music/controllers/SpotifyAppController.kt#L345